### PR TITLE
[BE]: Better cleanup optimized code from #151474

### LINKED ISSUE #151351: Sparse tensor conversion performance issues (CPU/GPU)
### 🐛 Describe the bug

I was investigating opportunities for improving "activation" sparsity as follows

```
import time
import torch

d = 1024 * 32
A = -torch.ones(d, d)
A[0, 0] = 111
A[10, 10] = 222     # most entries in A are < 0

T = torch.relu(A)         # materializes very sparse T as dense first
S = T.to_sparse_csr()     # would be nice to have something like S = torch.sparse_relu(A)
                          # but that is not the point of this bug yet
```

where I really would like to have a "sparsifying" relu to avoid materializing T as dense intermediate first.

However, while pondering on that, I noticed a huge performance (time and memory) difference between converting to COO or converting to CSR. Take the following annotated code

```
.. construct A..
time.sleep(10)      # INTERVAL 1
T = torch.relu(A)   # materializes very sparse T as dense first
time.sleep(10)      # INTERVAL 2
S = T.to_sparse()   # to COO
time.sleep(10)      # INTERVAL 3
```

and compare memory tracking for COO (to_sparse) with CSR (to_sparse_csr). Attached are the two plots. Both are running on CPU (but the problem occurs for GPU as well, same code path).

It is clear that COO behaves more or less as expected (first bump to get A, second bump to get T, then nothing more for S; again, my initial goal was to avoid the second bump for T, but read on). Then looking at CSR, we get a similar first bump to get A, second bump to T, but then a huge increase in memory to get to S. Also, the time increases substantially (notice the extra time in between INTERVAL 2 and INTERVAL 3).

Image

Image

### Versions

all versions

cc @alexsamardzic @nikitaved @pearu @cpuhrsch @amjames @bhosmer @jcaip

### PR DESCRIPTION
**Pull request by Skylion007**
This change addresses the first/second time/mem "spike" observed  Improves on #151474 by removing unnecessary stride calculations and unused arguments to the helper function

https://github.com/pytorch/pytorch/issues/151351

Fixes https://github.com/pytorch/pytorch/issues/151351

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM

**Comment by aartbik** [comment]
LGTM! Shall we add a TORCH_CHECK(not_zero_mask.dim() == 2) for our peace of mind?

**Review by eqy — APPROVED** [review]

**Comment by Skylion007** [comment]
> LGTM! Shall we add a TORCH_CHECK(not_zero_mask.dim() == 2) for our peace of mind?

Feel free to open a PR for that.
